documents/wg-application.md

```diff
@@ -0,0 +1,93 @@
+# CUDA WG - draft
+
+## What value do you want to bring to the project?
+
+The Working Group is an attempt to combine efforts to bring reliable support of writing safe GPGPU CUDA code with Rust.
+The work that will be done is supposed to clear the path for other GPGPU platforms in future.
```

Comment:
Sorry, I don't quite agree here. One of the main and probably the biggest topic we will discuss, is the safety in SIMT code. So far, without diging too much into the details, it seems, this will be nearly completely platform-agnostic.

All those compilation warnings and lints we will define, could be applied to any other "coming next" GPGPU platform.

Reply:
> One of the main and probably the biggest topic we will discuss, is the safety in SIMT code. 

This feels like something that ought to be done by the Unsafe Code Guidelines WG / Lang team, and only tangentially here.

> is the safety in SIMT code. So far, without diging too much into the details, it seems, this will be nearly completely platform-agnostic.

I don't know. Previous implementations of SIMT in Rust, [like this extension](https://github.com/rkruppe/spmd-vectorize-benchmarks), look quite different from what we will able to do for the time being in Rust without significantly extending the language. At the same time, this is all different from OpenMP, rayon, ISPC, and the other dozens different approaches to this.

Tackling this problem changes the roadmap of the WG from "getting CUDA support kind-of-working" to "language extensions for SIMT". 

While this is something that could be tackled later, I think it is unrealistic to try to tackle any of this at this point in time.

> All those compilation warnings and lints we will define, could be applied to any other "coming next" GPGPU platform.

Safe Rust code not invoking undefined behavior cannot be a lint or a warning. It has to be something that is impossible to do 100% of the time, and rejected by the type system.

Reply:
I am not saying that this issues are not important. Maybe it would be worth it to address them in a different question, e.g., about which kind of support we need from other teams, or which with other teams we expect to collaborate. We could add there that we expect to collaborate with the UCG WG and the Lang team to make sure that the SIMT programming model exposed by CUDA kernels remains sound or something like that.

Reply:
> Tackling this problem changes the roadmap of the WG from "getting CUDA support kind-of-working" to "language extensions for SIMT".

Okay, thanks, I see the point!